components/service/experiments/src/test/java/mozilla/components/service/experiments/ExperimentsTest.kt

```diff
@@ -30,25 +31,36 @@ import java.io.File
 import kotlin.reflect.full.functions
 import kotlin.reflect.jvm.isAccessible
 
+internal fun resetExperiments(
+    experimentSource: ExperimentSource,
+    experimentStorage: ExperimentStorage,
+    valuesProvider: ValuesProvider = ValuesProvider()
+) {
+    Experiments.overrideExperimentSource(experimentSource)
+    Experiments.overrideExperimentStorage(experimentStorage)
+    Experiments.overrideValuesProvider(valuesProvider)
+    Experiments.initialize(ApplicationProvider.getApplicationContext())
+}
+
 @RunWith(RobolectricTestRunner::class)
 class ExperimentsTest {
     @Test
     fun loadExperiments() {
         val experimentSource = mock(ExperimentSource::class.java)
         val experimentStorage = mock(ExperimentStorage::class.java)
-        val instance = Experiments(experimentSource, experimentStorage)
-        instance.loadExperiments()
+        resetExperiments(experimentSource, experimentStorage)
```

Comment:
I think `resetExperiments` and the related mocks could be moved to a `setup` function called before each test run. See [this](https://github.com/mozilla-mobile/android-components/blob/51ec2c968738b75786ee76c1e645819e4aa5e976/components/service/glean/src/test/java/mozilla/components/service/glean/storages/PingStorageEngineTest.kt#L46-L48). From the tests you could directly access the mocks via `Experiments.source` and `Experiments.storage`. This would make the tests *much* shorter and clearer, imho.

Reply:
I nee to set up the `source` & `storage` mocks before the setup functions happen.
I don't think that works out well here?

Reply:
We could set them up in `resetExperiments` as well, since they are always set up the same way, or provide default initializer for arguments in that function if you want to allow for custom ones.

Reply:
Ah, a default initializer is a good point!
They are not always set up the same way, so i can't always call `resetExperiments()` automatically from `@setup` code.

Reply:
I'm taking this to [bug 1536490](https://bugzilla.mozilla.org/show_bug.cgi?id=1536490) as well.